react-js/react-part-2.md

```diff
@@ -0,0 +1,175 @@
+
+- In the above instance of setState, an object has been passed as a parameter.
+- In case where previous state of React is used to update the new state(just like above instance), the correct way is to pass a function as a parameter which itself has previous state parameter and that can be used to update state.
+
+```javascript
+addValue = () => {
+  this.setState((prevState) => {
+    return {
+      value: prevState.value + 1
+    };
+  });
+};
+```
+
+For now let us use the object instance only.
+
+6. Similarly we will create a function which will decrement the value by 1.
+
+```javascript
+subtractValue = () => {
+  this.setState({
+    value: this.state.value - 1
+  });
+};
+```
+
+7. The functions have been created, we just need to associate the functions we have created to the buttons i.e. set the *onClick* attribute to the button which we have created.
+
+```html
+<button onClick="{this.addValue}">Increment</button>
+<button onClick="{this.subtractValue}">Decrement</button>
+```
+
+8. Let us extend this example a bit more by adding lower and upper limit to the value. If the value is _0_ then we cannot reduce the value and when the value becomes _10_, we should not be able to increment the value.  
+   This can be done by hiding the buttons on certain boundary value.
+
+- If the value is 0 then decrement button should not be displayed.
+- Similarly we have to hide the increment button when value becomes 10.  
+  **Note:- In React adding condition in JSX is done by two ways i.e. either using ternary operator( when there is a true part and a false part, `?` ) or by using double ampersand(for single value true, `&&` ).**  
+  So we add the conditions to buttons like :-
+
+```javascript
+{
+  this.state.value < 10 && 
+	<button onClick={this.addValue}>Increment</button>
+}
+{
+  this.state.value > 0 &&
+    <button onClick={this.subtractValue}>Decrement</button>
+}
+```
+
+That's all. We have completed our first application in React using React state. To see the complete code [Codepen](https://codepen.io/ashish24_nagpal/pen/jObzXzM).
+
+### Independent Practice
+
+For independent practice, let us create a simple game. Say you are a magician having a magic box and 3 random buttons. Out of 3 only 1 button is correct which reveals the text inside the magic box. Children will come to you and choose a button. If the button is correct, any random text should display which is present inside the box and status message should be displayed as _you win_. If the button chosen is incorrect then only message should be displayed as _you lose_.  
+UI should contain -
+
+- A div which is a magic box.
+- 3 buttons out of which only 1 is correct.
+- A hidden text which will only be displayed when correct button is clicked.
+- A message tag which display the status of the game whether the person has _won_ or _lost_ the game.
+- A reset button which will reset the state of the game.  
+  **_Hint :-_**
```

Comment:
what is the hint?  lines 168 and 169?  Fix the formatting so they appear as sub-bullets.